src/app/content/highlights/components/ConfirmationModal.tsx

```diff
@@ -1,47 +1,72 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
 import { FormattedMessage } from 'react-intl';
+import { useSelector } from 'react-redux';
 import styled from 'styled-components';
+import uuid from 'uuid';
 import Button from '../../../components/Button';
 import Modal from '../../../components/Modal';
 import { Body, BodyHeading, Footer } from '../../../components/Modal/styles';
-
-interface Props {
-  deny: () => void;
-  confirm: () => void;
-}
+import { assertDocument, assertNotNull } from '../../../utils';
+import { confirmationModalCallback, isConfirmationModalOpen } from '../selectors';
 
 // tslint:disable-next-line:variable-name
 const ConfirmationFooter = styled(Footer)`
   justify-content: space-between;
 `;
 
 // tslint:disable-next-line:variable-name
-const ConfirmationModal = ({deny, confirm}: Props) => {
-  return <Modal onModalClose={deny} heading='i18n:discard:heading'>
-    <Body>
-      <FormattedMessage id='i18n:discard:body'>
-        {(msg) => <BodyHeading>{msg}</BodyHeading>}
-      </FormattedMessage>
-    </Body>
-    <ConfirmationFooter>
-      <FormattedMessage id='i18n:discard:button:discard'>
-        {(msg) => <Button
-          data-testid='discard-changes'
-          onClick={confirm}
-          variant='primary'
+const ConfirmationModal = () => {
+  const isOpen = useSelector(isConfirmationModalOpen);
+  const callback = useSelector(confirmationModalCallback);
+  const document = assertDocument();
+  const modalNode = document.createElement('div');
+
+  modalNode.id = `dialog-${uuid()}`;
+  const root = assertNotNull(document.getElementById('root'), 'root element not found');
+
+  React.useEffect(() => {
+    if (isOpen) {
+      root.appendChild(modalNode);
+    }
+
+    return () => {
+      if (document.getElementById(modalNode.id)) {
+        root.removeChild(modalNode);
+      }
+    };
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen, modalNode, root]);
+
+  return ReactDOM.createPortal(
```

Comment:
Since now it is a normal component, I think you could just `return null` if it is not open and do not use portal